runtime/near-vm-errors/src/lib.rs

```diff
@@ -7,15 +7,18 @@ use std::fmt;
     Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
 )]
 pub enum VMError {
-    FunctionExecError(FunctionExecError),
-    // TODO: serialize/deserialize?
-    StorageError(Vec<u8>),
+    FunctionCallError(FunctionCallError),
+    /// Serialized external error from External trait implementation.
+    ExternalError(Vec<u8>),
```

Comment:
What is the reason we are serializing error here (CC @fckt )?

Reply:
The serialized error is part of RuntimeExt which comes from `primitives`. Either near-vm-logic has to depend on it, or we have to replicate the error. Right now the plan is to allow any error here but in serialized form.
Later we deserialize it back in the Runtime and get the original error type.